engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -2068,6 +2068,22 @@ public void updateRecoveryCheckpoints(PTOperator operator, UpdateCheckpointsCont
     if (checkpointGroup.size() > 1) {
       for (OperatorMeta om : checkpointGroup) {
         Collection<PTOperator> operators = plan.getAllOperators(om);
+        // Include unifiers in group if delay operator is present
+        // Do only if group contains a delay operator
```

Comment:
IMO if we have a dag like A -> B, where A is partitioned and A and B are in the same group, then the unifier should also be in the same group. Is this understanding correct?